runelite-client/src/main/java/net/runelite/client/plugins/PluginManager.java

```diff
@@ -215,12 +217,22 @@ public void startPlugins()
 		{
 			try
 			{
-				startPlugin(plugin);
+				SwingUtilities.invokeAndWait(() ->
+				{
+					try
+					{
+						startPlugin(plugin);
+					}
+					catch (PluginInstantiationException ex)
+					{
+						log.warn("Unable to start plugin {}", plugin.getClass().getSimpleName(), ex);
+						plugins.remove(plugin);
+					}
+				});
 			}
-			catch (PluginInstantiationException ex)
+			catch (InterruptedException | InvocationTargetException e)
 			{
-				log.warn("Unable to start plugin {}", plugin.getClass().getSimpleName(), ex);
-				plugins.remove(plugin);
+				throw new RuntimeException(e);
```

Comment:
Are you sure this is the correct way to handle this?

Reply:
this should never happen unless the thread gets interrupted (which should be unchecked anyway imo)